libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/MapboxDistanceFormatter.kt

```diff
@@ -208,7 +209,7 @@ class MapboxDistanceFormatter private constructor(
     }
 
     private fun Context.resourcesWithLocale(locale: Locale?): Resources {
-        val config = this.resources.configuration.also {
+        val config = Configuration(this.resources.configuration).also {
```

Comment:
@Guardiola31337 : this line fixes the issue that Notification's distance unit doesn't mach UI's distance unit.

this is the same fix as https://github.com/mapbox/mapbox-navigation-android/pull/3096/
we shouldn't update the current context's locale. It will impact the next run.